### ISSUE TITLE
DELETE /realms/:realm/orgs/:orgId takes ages (seconds)

### ISSUE THREAD

**Issue by mkyc** [p2-inc/keycloak-orgs#282]
When calling delete organization with [this method](https://phasetwo.io/api/delete-organization/) it takes about 2.5 seconds to delete single organization on my installation while any other operation takes milliseconds. Running it in parallel doesn't change that value very much, it just takes longer for each request to finish (10 threads mean that requests will take ~8 times longer). 

Connecting it with behavior that you cannot delete realm that contains organizations it makes any realm creation / deletion automation pretty hard if you have enough organizations (couple thousands in tests in my case). 

PH2 version: `quay.io/phasetwo/phasetwo-keycloak:24.0.5`

**Comment by xgp** [comment]
@mkyc How many organizations do you have? How many members per organization? How many identity providers?

I don't see the same behavior, but it may be due to different data.

**Comment by mkyc** [comment]
in example test realm I had: 
 - over 6k organizations
 - lets say about 200 members per organization (but it's hard to count for me right now)
 - 3 identity providers

**Comment by xgp** [comment]
Thanks for the detail. Is that 3 identity providers _per_ organization?

**Comment by mkyc** [comment]
Nope. Inside organization there is no identity provider. Those are on realm level. 

One more thing I can think of which might influence that behavior is that I have Infinispan cache defined between two nodes for HA.

**Comment by xgp** [comment]
Nothing happens related to orgs in infinispan. It is a strange behavior, as deleting one row, and a few hundred foreign key records shouldn't be a big deal. 

However, how often do you plan to actually be deleting orgs? I know it's not an ideal performance characteristic, but is it going to happen often?

**Comment by xgp** [comment]
@mkyc Thanks for the description. We're not really targeting such test use cases, so I doubt we will spend the time to debug this one. I'll leave it open if you have interest in trying to optimize the delete time. My guess is Hibernate is doing something naive like iterating through all the members and running an individual SQL DELETE. I'd suggest starting with running your tests with Hibernate set to DEBUG logging so you can see what's happening on a delete. Might be some JPA magic you could do to improve the performance.

**Comment by xkamil** [comment]
Is it possible to add bulk delete endpoint for organizations?
Now we have only DELETE /auth/realms/{realm}/orgs/{id} and it would be nice to also have DELETE /auth/realms/{realm}/orgs

**Comment by xkamil** [comment]
what about this code [link](https://github.com/p2-inc/keycloak-orgs/blob/44e5432b1454651873607a23b2f3e6d40fb3b0e5/src/main/java/io/phasetwo/service/resource/OrganizationResourceProviderFactory.java#L156) ?
I see that all organizations are removed when realm is removed. I added endpoint to call this via api locally + test and it works. I don't know how it behave with numbers like @mkyc mentioned above because in test it's hard to recreate this scenario.
If it's already there and it looks like it's used mayby it's worth exposing it via api so @mkyc can test it and provide us with more feedback on performance?
